lti_consumer/tests/unit/plugin/test_views_lti_ags.py

```diff
@@ -274,3 +301,556 @@ def test_create_lineitem_invalid_resource_link_id(self):
+                "pk": self.line_item.id
+            }
+        )
+
+    def test_retrieve_results(self):
+        """
+        Test the LTI AGS LineItem Score Creation.
+        """
+        self._set_lti_token('https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly')
+
+        # Create Score
+        response = self.client.get(self.results_endpoint)
+
+        self.assertEqual(response.status_code, 200)
+
+        # There should be 2 results (not include the empty score user's result)
+        self.assertEqual(len(response.data), 2)
+
+        # Check the data
+        primary_user_results_endpoint = reverse(
+            'lti_consumer:lti-ags-view-results',
+            kwargs={
+                "lti_config_id": self.lti_config.id,
+                "pk": self.line_item.id,
+                "user_id": self.primary_user_id
+            }
+        )
+        secondary_user_results_endpoint = reverse(
+            'lti_consumer:lti-ags-view-results',
+            kwargs={
+                "lti_config_id": self.lti_config.id,
+                "pk": self.line_item.id,
+                "user_id": self.secondary_user_id
+            }
+        )
+        self.assertEqual(
+            [dict(d) for d in response.data],
+            [
+                {
+                    "id": "http://testserver" + primary_user_results_endpoint,
+                    "scoreOf": "http://testserver" + self.lineitem_endpoint,
+                    "userId": self.primary_user_id,
+                    "resultScore": 83.0,
+                    "resultMaximum": 100.0,
+                    "comment": "This is exceptional work."
+                },
+                {
+                    "id": "http://testserver" + secondary_user_results_endpoint,
+                    "scoreOf": "http://testserver" + self.lineitem_endpoint,
+                    "userId": self.secondary_user_id,
+                    "resultScore": 25.0,
+                    "resultMaximum": 100.0,
+                    "comment": "This is not great work."
+                }
+            ]
+        )
+
+    def test_retrieve_results_for_user_id(self):
+        """
+        Test the LTI AGS LineItem Score Creation.
+        """
+        self._set_lti_token('https://purl.imsglobal.org/spec/lti-ags/scope/result.readonly')
+
+        results_user_endpoint = reverse(
+            'lti_consumer:lti-ags-view-results',
+            kwargs={
+                "lti_config_id": self.lti_config.id,
+                "pk": self.line_item.id,
+                "user_id": self.secondary_user_id
+            }
+        )
+
+        # Request results with userId
+        response = self.client.get(results_user_endpoint, data={"userId": self.secondary_user_id})
+
+        self.assertEqual(response.status_code, 200)
+
+        # There should be 1 result for that user
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['userId'], self.secondary_user_id)
```

Comment:
Don't we want to make an assertion about the result we got?

Reply:
@nedbat 

I'm not quite sure I understand what you mean. That line (841) makes an assertion that the result we received in the response, matches the user id requested. Combined with the check that there is only 1 result, that confirms that it filtered results with user ids that didn't match. There is already a test above that verifies the formatting of the individual results, and based on the records created during the `setUp()` for this class (see [here](https://github.com/edx/xblock-lti-consumer/blob/82af9de9e2509b9740cd45bfa376a3c6a837e616/lti_consumer/tests/unit/plugin/test_views_lti_ags.py#L711-L746)), there should only be 3 records, each with a unique user id. Thus, if the user id from the response matches the corresponding user id for the desired record (in this case it's `self.secondary_user_id`), then it's a sufficient match.

Should I add more assertions about the result and/or add comments to it?